platform/admin/core/admin-core-insecuredefaults/src/main/java/org/codice/ddf/admin/insecure/defaults/service/DefaultUsersDeletionScheduler.java

```diff
@@ -0,0 +1,234 @@
+import java.io.ObjectInputStream;
+import java.io.ObjectOutputStream;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.time.Duration;
+import java.time.Instant;
+import java.time.LocalDateTime;
+import java.time.ZoneId;
+import java.util.Collection;
+import java.util.stream.Stream;
+import org.apache.camel.CamelContext;
+import org.apache.camel.ServiceStatus;
+import org.apache.camel.builder.RouteBuilder;
+import org.apache.karaf.jaas.modules.BackingEngine;
+import org.apache.karaf.jaas.modules.BackingEngineFactory;
+import org.codice.ddf.configuration.AbsolutePathResolver;
+import org.osgi.framework.Bundle;
+import org.osgi.framework.BundleContext;
+import org.osgi.framework.FrameworkUtil;
+import org.osgi.framework.ServiceReference;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class DefaultUsersDeletionScheduler {
+
+  private static Path usersPropertiesFilePath =
+      Paths.get(new AbsolutePathResolver("etc/users.properties").getPath());
+  private static Path tempTimestampFilePath =
+      Paths.get(new AbsolutePathResolver("data/tmp/timestamp.bin").getPath());
+  private static final Logger LOGGER = LoggerFactory.getLogger(DefaultUsersDeletionScheduler.class);
+  private static final String ROUTE_ID = "deletionJob";
+  private final CamelContext context;
+
+  public DefaultUsersDeletionScheduler(CamelContext camelContext) {
+    this.context = camelContext;
+  }
+
+  public boolean scheduleDeletion() {
+    String cron = getCron();
+    if (cron == null) {
+      return false;
+    }
+    if (context != null && context.getRouteStatus(ROUTE_ID) != ServiceStatus.Started) {
+      try {
+        RouteBuilder builder =
+            new RouteBuilder() {
+              @Override
+              public void configure() throws Exception {
+                fromF("quartz2://scheduler/deletionTimer?cron=%s", cron)
+                    .routeId(ROUTE_ID)
+                    .bean(DefaultUsersDeletionScheduler.class, "removeDefaultUsers");
+              }
+            };
+        context.addRoutes(builder);
+        context.start();
+      } catch (Exception e) {
+        LOGGER.debug("Unable to create camel route.", e);
+        return false;
+      }
+    }
+    return true;
+  }
+
+  @VisibleForTesting
+  String getCron() {
+    if (!getTempTimestampFilePath().toFile().exists()) {
+
+      // Create temp file and add timestamp
+      Instant instant = Instant.now();
+      if (!createTempFile(instant)) {
+        return null;
+      }
+      return cronCalculator(instant);
+
+    } else {
+
+      // Read timestamp
+      Instant firstInstall = installationDate();
+      if (firstInstall == null) {
+        return null;
+      }
+
+      if (Instant.now()
+          .isAfter(
+              firstInstall
+                  .plus(Duration.ofDays(3))
+                  .minus(Duration.ofMinutes(30)))) { // Timestamp expired
+        removeDefaultUsers();
+        return null;
```

Comment:
✏️  Not necessary to change, but just a thought as I was perusing through:
Its seems odd to have the removeDefaultUsers side effect called inside of a method named `getCron` that returns a string. Seems like it could be a nice thing to split apart (or alternatively maybe this method just should be named) getCronOrDelete or something like that .